functional-tests/fn_bridge_withdraw_happy.py

```diff
@@ -0,0 +1,239 @@
+            }
+        )
+        web3.eth.send_transaction(
+            {
+                "from": el_address,
+                "to": BURNER_ADDRESS,
+                "value": balance_to_burn - (estimated_gas * gas_price),
+                "gas": estimated_gas,
+                "gasPrice": gas_price,
+            }
+        )
+        wait_until(lambda: int(rethrpc.eth_getBalance(el_address), 16) == 0)
+        # Ok now we have a clean state
+        self.logger.debug("Strata balance is zero")
+
+        # Get operators pubkey and musig2 aggregates it
+        bridge_pk = get_bridge_pubkey(seqrpc)
+        self.logger.debug(f"Bridge pubkey: {bridge_pk}")
+
+        seq_addr = seq.get_prop("address")
+        self.logger.debug(f"Sequencer Address: {seq_addr}")
+        bridge_pk = get_bridge_pubkey(seqrpc)
+
+        # Generate plenty of BTC to address
+        btcrpc.proxy.generatetoaddress(102, address)
+
+        # Deposit to the EL address
+        # NOTE: we need 2 deposits to make sure we have funds for gas
+        self.make_drt(ctx, el_address, bridge_pk)
+        self.make_drt(ctx, el_address, bridge_pk)
+        wait_until(lambda: int(rethrpc.eth_getBalance(el_address), 16) > 0)
+
+        # Get the balance of the EL address after the deposits
+        balance = int(rethrpc.eth_getBalance(el_address), 16)
+        self.logger.debug(f"Strata Balance after deposits: {balance}")
+        assert balance == 2 * DEPOSIT_AMOUNT * SATS_TO_WEI, "Strata balance is not expected"
+
+        # Send funds to the bridge precompile address for a withdrawal
+        change_address_pk = extract_p2tr_pubkey(withdraw_address)
+        self.logger.debug(f"Change Address PK: {change_address_pk}")
+        estimated_withdraw_gas = self.estimate_withdraw_gas(ctx, change_address_pk)
+        self.logger.debug(f"Estimated withdraw gas: {estimated_withdraw_gas}")
+        l2_tx_hash = self.make_withdraw(ctx, change_address_pk, estimated_withdraw_gas).hex()
+        self.logger.debug(f"Sent withdrawal transaction with hash: {l2_tx_hash}")
+
+        # Wait for the withdrawal to be processed
+        time.sleep(DEFAULT_BLOCK_TIME_SEC * 2)
+        tx_receipt = web3.eth.get_transaction_receipt(l2_tx_hash)
+        self.logger.debug(f"Transaction receipt: {tx_receipt}")
+        total_gas_used = tx_receipt["gasUsed"] * tx_receipt["effectiveGasPrice"]
+        self.logger.debug(f"Total gas used: {total_gas_used}")
+
+        # Make sure that the balance is expected
+        balance_post_withdraw = int(rethrpc.eth_getBalance(el_address), 16)
+        self.logger.debug(f"Strata Balance after withdrawal: {balance_post_withdraw}")
+        difference = DEPOSIT_AMOUNT * SATS_TO_WEI - total_gas_used
+        self.logger.debug(f"Strata Balance difference: {difference}")
+        assert difference == balance_post_withdraw, "balance difference is not expected"
+
+        # Mine blocks
+        btcrpc.proxy.generatetoaddress(12, UNSPENDABLE_ADDRESS)
+        time.sleep(1)
```

Comment:
Ideally this would be replaced by wait_until to indicate what exactly we are sleeping/waiting here for, and also to prevent longer waits if the condition meets early

Reply:
Done in a663e7424a18213078973394beca7685ab05d45f.